examples/lock_exchange.jl

```diff
@@ -0,0 +1,287 @@
+#  * Vertical closure CATKEVerticalDiffusivity handles small-scale vertical turbulence 
+#  * Weighted Essentially Non-Oscillatory (WENO) methods are useful for capturing sharp changes in density
+#  * ZStarCoordinate method allows for the top of the grid to move with the free surface
+#  * Runge Kutta method is good for integrating multiple processes 
+
+model = HydrostaticFreeSurfaceModel(; grid,
+    tracers = (:b, :e),      
+    buoyancy = BuoyancyTracer(),
+    closure = CATKEVerticalDiffusivity(),
+    momentum_advection = WENO(order=5), 
+    tracer_advection = WENO(order=7), 
+    vertical_coordinate = ZStarCoordinate(grid), 
+    free_surface = SplitExplicitFreeSurface(grid; substeps=10), 
+    timestepper = :SplitRungeKutta3
+)
+
+
+# ## Set Variable Density Initial Conditions 
+
+# Set initial conditions for lock exchange with different boyancies  
+bᵢ(x, z) = x > 4kilometers ? 0.01 : 0.06
+set!(model, b=bᵢ)
+
+
+# ## Defining Simulation Timestep 
+# 
+# Fast wave speeds make the equations stiff, so the CFL condition forces a small timestep Δt
+# when using explicit time-stepping to maintain numerical stability.
+
+# Set the timesteps 
+Δt = 1seconds 
+stop_time = 20hours
+simulation = Simulation(model; Δt, stop_time)
+
+# The TimeStepWizard helps ensure stable time-stepping with a (CFL) number of 0.7. Since the stability region 
+# of the numerical time stepper extends only up to CFL ≈ 1, we keep the CFL condition at a value 
+# of 0.7 to ensure robust and stable time stepping.
+
+clf_value = 0.7
+wizard = TimeStepWizard(cfl=clf_value, max_change=1.1, max_Δt=5.0)
+simulation.callbacks[:wizard] = Callback(wizard, IterationInterval(10))
+conjure_time_step_wizard!(simulation, cfl=clf_value)
+
+
+# ## Track Simulation Progress 
+
+# Wall clock represents the real world time as opposed to simulation time 
+wall_clock = Ref(time_ns())
+
+# Define callback function to log simulation iterations and time every 30 mins in simulation time 
+function progress(sim)
+    elapsed = 1e-9 * (time_ns() - wall_clock[])
+    msg = @sprintf("Iter: %7d, time: %s, wall: %s, max|w| = %6.3e m s⁻¹",
+                   iteration(sim),
+                   prettytime(sim),
+                   prettytime(elapsed),
+                   maximum(abs, sim.model.velocities.w))
+    wall_clock[] = time_ns()
+    @info msg
+    return nothing
+end
+
+save_interval = 30minutes
+add_callback!(simulation, progress, name = :progress, TimeInterval(save_interval))
+
+
+# ## Add Tracers and Diagnostics 
+
+# Here we define the fields that we want to save a snapshot of at every 30 minutes of simulation time. 
+# The JLD2Writer saves: 
+#  * b: buoyancy tracer
+#  * u: x component velocity (horizontal) 
+#  * u′: deviation of u from average of horizontal u
+#  * w: z component velocity (vertical)
+#  * N²: buoyancy stratification
+
+b = model.tracers.b
+e = model.tracers.e 
+u, v, w = model.velocities
+
+U = Field(Average(u)) 
+u′ = u - U            
```

Comment:
why do we subtract the mean U here? does it make sense in this example or is it leftover from the internal tide one?